cpp/arcticdb/version/local_versioned_engine.hpp

```diff
@@ -303,8 +303,12 @@ class LocalVersionedEngine : public VersionedEngine {
      * Get the queried, if specified, otherwise the latest, versions of index keys for each specified stream.
      * @param version_queries Only explicit versions are supported at the moment. The implementation currently
      * accepts deleted versions (e.g. to support reading snapshots) and it's the caller's responsibility to verify.
+     * A pair of std unordered maps are returned. The first one contains all the Atom keys for those queries that we 
```

Comment:
Wouldn't a union type make this clearer? Ie map keys that are `pair<StreamId, Version>` and `Version = VersionId | LatestVersion`

Reply:
Not applicable in this case. Latest version is still a VersionId, but it is a special case because it represents the latest and also because we don't know in advance which VersionId is (so we wouldn't be able to index the map, unless we create a new constant type representing the meaning of "latest_id", so we can use that as a key). It also would require to change the logic of the function producing it (get_stream_index_map). I widely discussed with William about it, and we decided to leave it as these changes are now, since we want to refactor all this part eventually.